vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java

```diff
@@ -370,8 +371,15 @@ private void resetContent() {
 
     private void onTargetAttach(UI ui) {
         ui.getInternals().addComponentDependencies(ContextMenu.class);
-        ui.getPage().executeJs(
-                "window.Vaadin.Flow.contextMenuConnector.init($0)", target);
+        if (targetBeforeClientResponseRegistration == null) {
+            targetBeforeClientResponseRegistration = ui
```

Comment:
This should be removed and set to null in `setTarget`, to cancel the `connector.init` call for the previous target in case the target is changed multiple times during the same roundtrip.